backend/src/main/java/com/nikogrid/backend/controllers/AuthenticationController.java

```diff
@@ -0,0 +1,100 @@
+package com.nikogrid.backend.controllers;
+
+import com.nikogrid.backend.auth.JwtGenerator;
+import com.nikogrid.backend.auth.SecurityConfig;
+import com.nikogrid.backend.dto.AuthResponse;
+import com.nikogrid.backend.dto.LoginDTO;
+import com.nikogrid.backend.dto.RegisterDTO;
+import com.nikogrid.backend.services.AuthenticationService;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.server.Cookie.SameSite;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/auth")
+@Validated
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final AuthenticationManager authenticationManager;
+    private final JwtGenerator jwtGenerator;
+    private final boolean secureCookies;
+
+    @Autowired
+    public AuthenticationController(
+            AuthenticationService authenticationService,
+            AuthenticationManager authenticationManager,
+            JwtGenerator jwtGenerator,
+            @Value("${jwt.secure-cookies:false}") boolean secureCookies
+    ) {
+        this.authenticationService = authenticationService;
+        this.authenticationManager = authenticationManager;
+        this.jwtGenerator = jwtGenerator;
+        this.secureCookies = secureCookies;
+    }
+
+    @PostMapping("/login")
+    @PreAuthorize("!isAuthenticated()")
+    @ResponseStatus(HttpStatus.OK)
```

Comment:
```suggestion
    @ResponseStatus(HttpStatus.OK)
```
This is still a problem